python/paddle/fluid/dygraph/dygraph_to_static/program_translator.py

```diff
@@ -391,25 +359,19 @@ def func(x):
         assert callable(
             dygraph_func
         ), "Input dygraph_func is not a callable in ProgramTranslator.get_output"
-        if in_dygraph_mode() or not self.enable_declarative:
+        if not self.enable_declarative:
             logger.info(
-                "The ProgramTranslator.get_output doesn't work in dygraph "
-                "mode or set ProgramTranslator.enable to False. We will "
-                "just return dygraph output.")
+                "The ProgramTranslator.get_output doesn't work when setting ProgramTranslator.enable = False. "
+                "We will just return dygraph output.")
             return dygraph_func(*args, **kwargs)
 
-        program_cache = self.get_program_cache()
-        outputs = program_cache.build_program_and_return_output(dygraph_func,
-                                                                *args, **kwargs)
-        if not program_cache.in_build_process:
-            outputs = self._run(*args, **kwargs)
-            with guard():
-                outputs = map_structure(to_variable, outputs)
-                if len(outputs) == 1:
-                    outputs = outputs[0]
-                else:
-                    outputs = tuple(outputs)
-        return outputs
+        function_spec = FunctionSpec(dygraph_func, args, kwargs)
+        _, partial_program_layer = self._program_cache[function_spec]
+
+        if args and isinstance(args[0], layers.Layer):
+            args = args[1:]
+
+        return partial_program_layer(args)
```

Comment:
if user change`train` or `eval` mode when added @declarative, how to change partial_program_layer‘s mode?

Reply:
Thanks, I I'm adding this feature...

Reply:
Done~~